app/client/src/features/accounts/UserSideNav.tsx

```diff
@@ -88,11 +90,21 @@ export function UserSideNavLoader() {
 }
 
 export function UserSideNav({ queryRef, onClick }: UserSideNavProps) {
+    const data = usePreloadedQuery<UserSideNavQuery>(USER_SIDE_NAV_QUERY, queryRef);
+
     const classes = useStyles();
     const router = useRouter();
     usePreloadedQuery(USER_SIDE_NAV_QUERY, queryRef);
     const [selected, setSelected] = React.useState<Keys | undefined>(findTab(router.pathname));
 
+    const { displaySnack } = useSnack();
+    React.useEffect(() => {
+        if (!data.isOrganizer && router.pathname === '/organizations/me') {
+            router.push('/dashboard');
+            displaySnack('Organizers-only path!');
+        }
+    }, [data.isOrganizer, displaySnack, router]);
+
```

Comment:
I think this would be better delegated to a component.  Take a look at https://github.com/c2c-project/prytaneum/blob/dev/app/client/src/components/ConditionalRender.tsx

Maybe you could make a version of this that's for the user type? 

```tsx
<RoleGuard organizer>
  <SomeOrganizerOnlyComponent />
</RoleGuard>
<RoleGuard organizer moderator>
   // An organizer or a moderator could access inside of here
  <SomeComponent />
</RoleGuard>
<Roleguard organizer redirect="/path/to/go/to/if/not/an/organizer">
  <SomethingSensitive />
</RoleGuard>
```